File: modules/uCommon/app/models/Company.scala
package models

import org.springframework.data.neo4j.annotation._
import org.springframework.data.neo4j.support.index._

@NodeEntity
class Company {

  @GraphId
  var id : java.lang.Long = _

  var name : String = _
  var street: String = _
  var city: String = _
  var state: String = _
  var postal: String = _
  var country: String = _

}

File: modules/uCommon/app/neo4j/models/Company.java
package neo4j.models;

import java.util.Set;

import org.neo4j.graphdb.Direction;
import org.neo4j.graphdb.RelationshipType;
import org.springframework.data.neo4j.annotation.Fetch;
import org.springframework.data.neo4j.annotation.GraphId;
import org.springframework.data.neo4j.annotation.Indexed;
import org.springframework.data.neo4j.annotation.NodeEntity;
import org.springframework.data.neo4j.annotation.RelatedTo;

@NodeEntity
public class Company extends AbstractNode {

  @Indexed
  public String name;

  public Company(String name) {
    this.name = name;
  }

  public Company() {
  }

  @Override
  public String toString() {
    return String.format("Company{name='%s'}", name);
  }

}
